### ISSUE TITLE
Why mmseg uses IterBasedRunner rather than EpochBasedRunner

### ISSUE THREAD

**Issue by xpngzhng** [open-mmlab/mmsegmentation#76]
I found several inconsistencies in mmcls, mmseg and mmdet
while mmcls and mmdet use EpochBasedRunner, mmseg uses IterBasedRunner
I want to know why mmseg is designed like this, thx

**Comment by xvjiarui** [comment]
Hi @xpngzhng 
Iteration based training is more flexible than epoch based, which is also the practice in the field, e.g. [CSAIL semantic segmentation](https://github.com/CSAILVision/semantic-segmentation-pytorch), [torchcv](https://github.com/donnyyou/torchcv). 
The learning rate schedule is also iteration based.

**Comment by YimianDai** [comment]
@xvjiarui why iteration based training is more flexible than epoch based?

**Comment by YimianDai** [comment]
Many thanks!

**Comment by hiyyg** [comment]
> By epoch, you could only save upon finishing one epoch. By iteration, you could save any step you like. 

Are there any other advantages of using iter based runner against epoch based runner?
